### ISSUE TITLE
Upgrade to 4.2?

### ISSUE THREAD

**Issue by TotemChris** [realm/realm-mobile-platform#35]
I just tried upgrading my RHEL instance running 2.3 to 3.0, but to my surprise I received 1.0.0_BETA_4.2-336.el7, along with a realm-object-server service that wouldn't start, even after reboot of the server.  Just dropping a note here because I didn't see any release notes beyond 3.0 yet.

**Comment by dannygrob** [comment]
No replies? 
I can not start 1.0.0-BETA-4.2-336 as well as 1.0.0-BETA-3.0-321 on Ubuntu, and can't find anything in my logs about why. How to find out what's going wrong? If I downgrade to 1.0.0-BETA-2.3-310 the service just starts again. (sudo apt-get install realm-object-server-de=1.0.0-BETA-2.3-310)

I have a lot of users complaining about the server not syncing, and to me the service syncs only 1 time in the beginning. I don't have any logs or indication what the issues are, but I'm keeping a close eye here.

**Comment by teotwaki** [comment]
Hi,

We're in the middle of a company-wide meeting this week. We're looking into this, and will get back to you ASAP.

**Comment by dannygrob** [comment]
Thanks. After every upgrade from 2.3 the service won't start. I did some tests, and removing the auth.realm resolves this, and the service will start. If i keep the old auth it results in 'segmentation fault' when starting up manually:
`/usr/bin/realm-object-server -c /etc/realm/configuration.yml`
`Segmentation fault`

However, by doing this I'll loose all my users...

**Comment by teotwaki** [comment]
Yeah, we've identified the cause of this. We're working on a tool that fixes this. For non-production users, the obvious solution is to just remove the auth realm and be happy. We know this is very impacting, so it's high on our priority list.

**Comment by dannygrob** [comment]
Hi,

Any updates regarding this tool about when do you expect to release this?
Thanks for taking your time to look into it!

Danny

**Comment by aalfath** [comment]
+1

I'm also having this problem so it would be great to know about any updates.

**Comment by teotwaki** [comment]
Working on this as we speak. Should be available in the next release. Will keep you posted. Sorry for the lack of delay, it's been a crazy couple of weeks.

PS: With such short sentences, I almost felt like I was writing a telegram :smiley_cat:.

**Comment by dannygrob** [comment]
Is this fixed with the recent updates (4.2.2 and 4.2.1)?

**Comment by teotwaki** [comment]
This should be fixed in our next release. I'll close this when it's the case.

**Comment by dannygrob** [comment]
I'm sorry to ask about it again, but this is a very essential fix for me. I'm answering 20-30 support e-mails a day. Is there any planning for a release date that I can communicate to my users? Or if it takes long, the possibility to release this tool seperatly?

**Comment by teotwaki** [comment]
Did you try the latest release? I'm not in the office so can't check right now, but I was under the impression this had been fixed.

**Comment by dannygrob** [comment]
I'm getting this after updating:

```
Error: The following changes cannot be made in additive-only schema mode:
- Primary Key for class 'RealmFile has been added.
- Primary Key for class 'Token has been added.
- Primary Key for class 'User has been added.
    at Error (native)
    at Manager.start (/usr/lib/nodejs/realm-object-server/src/node/services/auth/manager.js:23:18)
    at new Manager (/usr/lib/nodejs/realm-object-server/src/node/services/auth/manager.js:17:10)
    at new Service (/usr/lib/nodejs/realm-object-server/src/node/services/auth/service.js:34:21)
    at new Server (/usr/lib/nodejs/realm-object-server/src/node/services/server.js:43:24)
    at Object.<anonymous> (/usr/lib/nodejs/realm-object-server/src/node/index.js:151:16)
    at Module._compile (module.js:410:26)
    at Object.Module._extensions..js (module.js:417:10)
    at Module.load (module.js:344:32)
    at Function.Module._load (module.js:301:12)
```

**Comment by teotwaki** [comment]
Right. I'll get the tool in your hands tomorrow or Tuesday. Can you email at help@realm.io, mention my name, so we can take this in private?

**Comment by teotwaki** [comment]
@dannygrob ^

**Comment by raudabaugh** [comment]
Could you please consider providing the migration tool to everyone? I think this is a widespread issue.

**Comment by teotwaki** [comment]
The code will be included in the startup sequence of the server, so it an upgrade will fix it for everyone.

**Comment by dannygrob** [comment]
Is the startup script working for anyone? For me it doesn't fix it. I don't have the tool yet either so I'm still stuck

**Comment by bigfish24** [comment]
@teotwaki or @sorenvind can you clarify?

**Comment by teotwaki** [comment]
Sorry, this is being coordinated via help@realm.io, so I lost track of this one. The current version doesn't include the migration stuffs just yet, that will be for the next release. I'm currently trying to extract the migration into a CLI tool for Danny.

**Comment by teotwaki** [comment]
This has been fixed in the 1.0 release.
